components/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/handler/grant/AccountSwitchGrantHandler.java

```diff
@@ -99,9 +119,19 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
 
             return false;
         }
+        AuthenticatedUser authenticatedUser = OAuth2Util.getUserFromUserName(associatedUser.toFullQualifiedUsername());
+        if (isSubOrganizationUser(accessTokenDO)) {
+            authenticatedUser.setAccessingOrganization(accessTokenDO.getAuthzUser().getAccessingOrganization());
+            authenticatedUser.setUserResidentOrganization(accessTokenDO.getAuthzUser().getUserResidentOrganization());
+            authenticatedUser.setFederatedUser(true);
+            authenticatedUser.setFederatedIdPName(accessTokenDO.getAuthzUser().getFederatedIdPName());
+            authenticatedUser.setTenantDomain(accessTokenDO.getAuthzUser().getTenantDomain());
+            String userId = getUserId(username, tenantDomain);
+            authenticatedUser.setUserId(userId);
+            authenticatedUser.setUserName(userId + "@" + accessTokenDO.getAuthzUser().getUserResidentOrganization());
```

Comment:
what's the reason to change these?

Reply:
If these not changed default values set as null. token request fails